### ISSUE TITLE
Crash on trying to open solution from start window

### ISSUE THREAD

**Issue by mrward** [mono/monodevelop#7659]
Build Information
Release ID: 801002578
Git revision: a79542bdc4aba2fe7ea8589b8bff6ea239853078
Build date: 2019-05-24 08:54:27+00
Build branch: release-8.1
Xamarin extensions: b894e02a8f68200720b84feb210215a3c1c2dc66

Clicking a solution on the start window terminates VS Mac.

 ```
FATAL ERROR [2019-05-24 11:37:54Z]: Visual Studio failed to start. Some of the assemblies required to run Visual Studio (for example gtk-sharp)may not be properly installed in the GAC.
System.ObjectDisposedException: Cannot access a disposed object.
Object name: 'GetToCodeMainPageView'.
  at Foundation.NSObject.get_SuperHandle () [0x0005b] in /Library/Frameworks/Xamarin.Mac.framework/Versions/5.6.0.25/src/Xamarin.Mac/Foundation/NSObject2.cs:458 
  at AppKit.NSView.RegisterForDraggedTypes (System.String[] newTypes) [0x00047] in /Library/Frameworks/Xamarin.Mac.framework/Versions/5.6.0.25/src/Xamarin.Mac/AppKit/NSView.g.cs:4148 
  at Xamarin.Ide.GetToCode.GetToCodeMainPageView.ViewDidMoveToSuperview () [0x00000] in /Users/vsts/agent/2.150.3/work/1/s/md-addins/Xamarin.Ide/Xamarin.Ide/Xamarin.Ide.GetToCode/GetToCodeMainPageView.cs:195 
  at (wrapper managed-to-native) Gtk.Application.gtk_main()
  at Gtk.Application.Run () [0x00001] in /Users/builder/jenkins/workspace/build-package-osx-mono/2018-08/external/bockbuild/builds/gtk-sharp-None/gtk/Application.cs:145 
  at MonoDevelop.Ide.IdeStartup.Run (MonoDevelop.Ide.MonoDevelopOptions options) [0x00455] in /Users/vsts/agent/2.150.3/work/1/s/monodevelop/main/src/core/MonoDevelop.Ide/MonoDevelop.Ide/IdeStartup.cs:221 
  at MonoDevelop.Ide.IdeStartup.Main (System.String[] args, MonoDevelop.Ide.Extensions.IdeCustomizer customizer) [0x0009e] in /Users/vsts/agent/2.150.3/work/1/s/monodevelop/main/src/core/MonoDevelop.Ide/MonoDevelop.Ide/IdeStartup.cs:729 
```

> VS bug [#899164](https://devdiv.visualstudio.com/DevDiv/_workitems/edit/899164)

**Comment by netonjm** [comment]
I think the problem is we need to check if the window is disposing and then return; cc @jsuarezruiz

**Comment by xamarin-release-manager** [comment]
> [sync] [VS-20] Comment by **xamarinc**



Fixed in version 8.3.0.86 (master)



Pull Request #4688 merged by: Vsevolod Kukol  
 Author: xamarin  
 Commit: 01fd0616adee270ee4059f7028b77692ee84f6e2 (xamarin/md-addins)  
 Included in Commit: 5f119be329ab7af9e00dfe42da7de5c2b7c091eb (mono/monodevelop)

**Comment by xamarin-release-manager** [comment]
> [sync] [VS-21] Comment by **xamarinc**



Fixed in version 8.1.0.2587 (release-8.1)



Pull Request #4690 merged by: Vsevolod Kukol  
 Author: xamarin  
 Commit: b9277d1dab85de7282208377f441075a202daefd (xamarin/md-addins)  
 Included in Commit: f8a2520e516401bd4e9deefda4d852b57e76934a (mono/monodevelop)

**Comment by xamarin-release-manager** [comment]
> [sync] [VS-27] Comment by **xamarinc**



Fixed in version 8.2.0.542 (release-8.2)



Pull Request #4688 merged by: Vsevolod Kukol  
 Author: xamarin  
 Commit: 01fd0616adee270ee4059f7028b77692ee84f6e2 (xamarin/md-addins)  
 Included in Commit: 1036aa86dce31b880dab3d14cd1214d3c2e5046f (mono/monodevelop)
